### PR TITLE
feat: add data model for appointments

### PR DESCRIPTION
**Pull request by timidak**
- updated lecture entity
- updated match entity
- created migration

### CODE REVIEW

File: common/entity/Lecture.ts [hunk 1]
```diff
@@ -33,4 +42,54 @@ export class Lecture {
...
+        array: true,
+    })
+    participants: User['userID'][];
+
+    @Column({
+        type: 'text',
+        default: [],
+        array: true,
+    })
+    declinedBy: User['userID'][];
```
Comment (Jonasdoubleyou) [inline common/entity/Lecture.ts:94]: Is this participants or instructors? Is it only possible to decline or accept?

File: common/migration/1682408426422-appointments.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import {MigrationInterface, QueryRunner} from "typeorm";
+
+export class appointments1682408426422 implements MigrationInterface {
+    name = 'appointments1682408426422';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        await queryRunner.query(`ALTER TABLE "lecture" DROP COLUMN "meetingLink"`);
```
Comment (Jonasdoubleyou) [inline common/migration/1682408426422-appointments.ts:7]: ?

File: common/migration/1682408426422-appointments.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "organizers" text array NOT NULL DEFAULT '{}'`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "participants" text array NOT NULL DEFAULT '{}'`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "declinedBy" text array NOT NULL DEFAULT '{}'`);
+        await queryRunner.query(`ALTER TABLE "pupil" ALTER COLUMN "lastTimeCheckedNotifications" SET DEFAULT '"1970-01-01T00:00:00.000Z"'`);
```
Comment (Jonasdoubleyou) [inline common/migration/1682408426422-appointments.ts:11]: ?

File: common/migration/1682408426422-appointments.ts (hunk 2 above)
Comment (timidak) [reply common/migration/1682408426422-appointments.ts:7]: My bad.. my local database had some artifacts. 😅

File: common/migration/1682414667095-Appointments_data.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "title" text`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "description" text`);
+        await queryRunner.query(`CREATE TYPE "public"."lecture_appointmenttype_enum" AS ENUM('group', 'match', 'internal', 'legacy')`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "appointmentType" "public"."lecture_appointmenttype_enum" NOT NULL DEFAULT 'legacy'`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "isCanceled" boolean DEFAULT false`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "organizers" text array NOT NULL DEFAULT '{}'`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "participants" text array NOT NULL DEFAULT '{}'`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "declinedBy" text array NOT NULL DEFAULT '{}'`);
+        await queryRunner.query(`ALTER TABLE "lecture" ADD "matchId" integer`);
+        await queryRunner.query(`ALTER TABLE "pupil" ALTER COLUMN "lastTimeCheckedNotifications" SET DEFAULT '"1970-01-01T00:00:00.000Z"'`);
```
Comment (Jonasdoubleyou) [inline common/migration/1682414667095-Appointments_data.ts:16]: hm, still

**Review by Jonasdoubleyou — DISMISSED** [review]

**Review by Jonasdoubleyou — APPROVED** [review]

**Comment by timidak** [comment]
Should be ready now, I added the public fields in authorizations. The next PR will include the new role and the enhanced accessCheck. @Jonasdoubleyou

**Review by Jonasdoubleyou — APPROVED** [review]

**Pull request merged**